src/test/java/io/jenkins/plugins/oidc_provider/IdTokenCredentialsTest.java

```diff
@@ -170,6 +170,93 @@ public class IdTokenCredentialsTest {
         });
     }
 
+    @Test public void requiredEnvironmentVariablesAreHonoured() throws Throwable {
+        rr.then(r -> {
+            IdTokenStringCredentials c = new IdTokenStringCredentials(CredentialsScope.GLOBAL, "test", null);
+            CredentialsProvider.lookupStores(r.jenkins).iterator().next().addCredentials(Domain.global(), c);
+            IdTokenConfiguration cfg = IdTokenConfiguration.get();
+            cfg.setClaimTemplates(Collections.singletonList(new ClaimTemplate("ok", "true", new BooleanClaimType())));
+            cfg.setGlobalClaimTemplates(Collections.singletonList(new ClaimTemplate("sub", "jenkins", new StringClaimType())));
+
+            cfg.setBuildClaimTemplates(Arrays.asList(
+                new ClaimTemplate("sub", "${JOB_NAME}", new StringClaimType(), "JOB_NAME"),
+                new ClaimTemplate("sub", "${JOB_NAME}:pull_request", new StringClaimType(), "JOB_NAME BUILD_ID"),
```

Comment:
Confusing to use a variable which will actually always be set—`BUILD_ID`—to test a behavior (checking for PRs) which would normally be using another variable—`CHANGE_ID`.